src/lib/classes/styling/ZodSchemas/AttributeSchemas/ColorAttribute.ts

```diff
@@ -0,0 +1,30 @@
+import { z } from "zod";
+
+/**
+ * The values for the colors must be within the range of 0-1 as the implementation is based on the color() css function.
+ * See information about the color function on: https://developer.mozilla.org/en-US/docs/Web/CSS/color_value/color.
+ * The alpha value is optional, and is also a number within the range 0-1.
+ *
+ * Example:
+ * 		To represent the color "rgb(255 0 0)"" it is writen as "color(srgb 1 0 0)".
+ * 		To convert the individual RGB values from the range 0-255 to the range 0-1 one simply has to use the following formula:
+ * 			(old value)/255 = new value
+ */
+const ColorWithAlpha = z.tuple([
+	z.string(), // The Color Gamut
+	z.number(), // Red value (Range 0-1)
```

Comment:
```suggestion
	z.number().gte(0).lte(1), // Red value
```
You should do this for all numbers, this would also  allow you to remove the `outOfColorRange` check.

I would probably extract it into a separate type called something like `ColorRange` and then use that as the type for each entry in the tuple :)

Docs: https://zod.dev/?id=numbers